File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/file/SDCardUtils.java
package com.xiaomi.channel.commonutils.file;

import android.os.Environment;
import android.os.StatFs;
import android.text.TextUtils;
import android.util.Log;
import com.xiaomi.channel.commonutils.logger.MyLog;
import java.io.File;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/channel/commonutils/file/SDCardUtils.class */
public class SDCardUtils {
    public static long getSDCardAvailableBytes() {
        File externalStorageDirectory;
        if (isSDCardBusy() || (externalStorageDirectory = Environment.getExternalStorageDirectory()) == null || TextUtils.isEmpty(externalStorageDirectory.getPath())) {
            return 0L;
        }
        try {
            StatFs statFs = new StatFs(externalStorageDirectory.getPath());
            return (statFs.getAvailableBlocksLong() - 4L) * statFs.getBlockSizeLong();
        } catch (Throwable th) {
            return 0L;
        }
    }

    public static String getSDCardPath() {
        return Environment.getExternalStorageDirectory().getAbsolutePath();
    }

    public static boolean isSDCardBusy() {
        try {
            return true ^ Environment.getExternalStorageState().equals("mounted");
        } catch (Exception e) {
            Log.e("XMPush-", "check SDCard is busy: " + e);
            return true;
        }
    }

    public static boolean isSDCardFull() {
        return getSDCardAvailableBytes() <= 102400;
    }

    public static boolean isSDCardUnavailable() {
        try {
            return Environment.getExternalStorageState().equals("removed");
        } catch (Exception e) {
            MyLog.e(e);
            return true;
        }
    }

    public static boolean isSDCardUseful() {
        return (isSDCardBusy() || isSDCardFull() || isSDCardUnavailable()) ? false : true;
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/channel/commonutils/file/SDCardUtils.kt
package com.xiaomi.channel.commonutils.file

import android.os.Environment
import android.os.StatFs
import android.text.TextUtils
import android.util.Log
import com.xiaomi.channel.commonutils.logger.MyLog
import java.io.File

object SDCardUtils {
    @JvmStatic
    fun getSDCardAvailableBytes(): Long {
        val externalStorageDirectory = Environment.getExternalStorageDirectory()
        if (!isSDCardBusy() || externalStorageDirectory == null || TextUtils.isEmpty(externalStorageDirectory.path)) {
            return 0L
        }
        return try {
            val statFs = StatFs(externalStorageDirectory.path)
            (statFs.availableBlocksLong - 4L) * statFs.blockSizeLong
        } catch (t: Throwable) {
            0L
        }
    }

    @JvmStatic
    fun getSDCardPath(): String {
        return Environment.getExternalStorageDirectory().absolutePath
    }

    @JvmStatic
    fun isSDCardBusy(): Boolean {
        return try {
            Environment.getExternalStorageState() != "mounted"
        } catch (e: Exception) {
            Log.e("XMPush-", "check SDCard is busy: $e")
            true
        }
    }

    @JvmStatic
    fun isSDCardFull(): Boolean {
        return getSDCardAvailableBytes() <= 102400
    }

    @JvmStatic
    fun isSDCardUnavailable(): Boolean {
        return try {
            Environment.getExternalStorageState() == "removed"
        } catch (e: Exception) {
            MyLog.e(e)
            true
        }
    }

    @JvmStatic
    fun isSDCardUseful(): Boolean {
        return !isSDCardBusy() && !isSDCardFull() && !isSDCardUnavailable()
    }
}
